### PR TITLE
Transition code labeling

### PR DESCRIPTION
**Pull request by zelosos**
constants from gameplay that belong to GB HW have been moved

### CODE REVIEW

File: src/code/bank20.asm [hunk 1]
```diff
@@ -1857,70 +1857,95 @@ func_020_54F5::
...
+.right:  db $00
+.left:   db $00
+.up:     db $03
+.down:   db $00
+
+RegionOriginLowOffset::
+.right:  db $01
+.left:   db $1F
+.up:     db $E0
+.down:   db $20
```
Comment (zelosos) [inline src/code/bank20.asm:1884]: .label: or .label here?

Comment (kemenaran) [reply src/code/bank20.asm:1884]: I'd say `.label:` (it seems the form without the semicolon is going to be deprecated in rgbds soon).

Comment (zelosos) [reply src/code/bank20.asm:1884]: ok then i will change it at the other place

File: src/constants/gbhw.asm [hunk 2]
```diff
@@ -140,3 +140,19 @@ rPCM12      EQU $ff76 ; (00h) - Always 00h (Read Only)
...
+; display height in pixel
+DISPLAY_HEIGHT      equ 144
+; pixel widht of one tile
+TILE_WIDTH          equ 8
+; pixel height of one tile
+TILE_HEIGHT         equ 8
+; number of tiles in one row
+TILES_PER_ROW       equ DISPLAY_WIDTH/(TILE_WIDTH*2)
+; number of tiles in one coloumn
+TILES_PER_COLOUMN   equ DISPLAY_HEIGHT/(TILE_HEIGHT*2)
```
Comment (kemenaran) [inline src/constants/gbhw.asm:154]:
```suggestion
; number of tiles in one column
TILES_PER_COLUMN    equ DISPLAY_HEIGHT/(TILE_HEIGHT*2)
```

File: src/constants/vfx.asm [hunk 3]
```diff
@@ -22,3 +22,6 @@ GROUND_VFX_NONE          equ $00
 GROUND_VFX_SHALLOW_WATER equ $05
 GROUND_VFX_GRASS         equ $06
 GROUND_VFX_PIT           equ $50
+
+; height of the window on the top of the screen
+WINDOW_HEIGHT equ 2 ; number in tiles
```
Comment (kemenaran) [inline src/constants/vfx.asm:27]: `vfx.asm` is intended to hold constants for small visual effects rendered using sprites.
Wouldn't this constant be more in place in, say, `constants/gfx.asm` ?

**Review by kemenaran — APPROVED** [review]
Thanks for your work. Since the beginning I'm especially interested in the camera and transition code: I started contributed to this disassembly because I wanted to mod the game with a continuous scrolling camera (instead of screen-by-screen transition).

Unfortunately this is probably not possible (or at least a huuge work), but it' still nice to understand precisely how the existing camera code works.

File: src/constants/gbhw.asm (hunk 2 above)
Comment (zelosos) [reply src/constants/gbhw.asm:154]: needs to be changed in the usage as well

File: src/constants/vfx.asm (hunk 3 above)
Comment (zelosos) [reply src/constants/vfx.asm:27]: I did not understood the difference between vfx and gfx

**Comment by zelosos** [comment]
So you are not interested in a fully disassembly of the game?

**Comment by zelosos** [comment]
Do you mean a transition like the one used in the Pokemon games?

File: src/code/bank20.asm (hunk 1 above)
Comment (daid) [reply src/code/bank20.asm:1884]: Local `.label` is not going to be deprecated, only `label`. But I would always add semicolons to labels just to be clear.

Comment (zelosos) [reply src/code/bank20.asm:1884]: `.label` is going to be depricated? So no local labels anymore?
What do you mean with semicolons?
.label ; comment
or
; comment
.label
or did you meant colon (:) ?

Comment (zelosos) [reply src/code/bank20.asm:1884]: done

File: src/constants/vfx.asm (hunk 3 above)
Comment (kemenaran) [reply src/constants/vfx.asm]: VFX is for Visual Effects, and is specifically around visuals produced using small sprites (especially around Link's feet, like water effects or dust smoke, but not only).

GFX is for Graphics (or Graphic Effetcs), and is everything else: graphic elements, backgrounds, etc.

So gfx.asm is the most generic constants file; while vfx.asm is very specific to one single purpose.

**Comment by kemenaran** [comment]
> So you are not interested in a fully disassembly of the game?

I definitely am interested in a full disassembly of the game :) But my initial interest was specifically focused on the camera code (even though it expanded to the whole disassembly project later).

> Do you mean a transition like the one used in the Pokemon games?

Exactly, where the whole world is loaded seamlessly.

To be honest, I don't think a seamless overworld can be achieved on Zelda DX code. So I thought:
- Maybe the Overworld can have a continuous scrolling, but Dungeons rooms could still load room-by-room (like the Switch remake does).
- Maybe the Overworld could do continuous scrolling only on 2x2 areas, and then do a transition (somehow like Zelda SNES does). It would avoid the need to switch tilesets (as tilesets are only changed every 2 rooms), and the need to load many entities at once (only the entities for 4 rooms would be loaded at once).

In the end, the main challenge would be to increase the number of visible entities on screen – as the need for WRAM, Tiles RAM and OAM RAM would increase a lot. Even when using the extra WRAM banks, it would still be a challenge. But maybe, mayyyybe it could be done :)

**Pull request merged**